From 87b04847bb246fb470d2a99d9b5f12eedb221dcf Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Thu, 25 Oct 2012 17:49:00 -0600
Subject: [PATCH 018/248] FogBugz #81648: Clean up SMP for SOCFPGA

The SMP implementation for SOCFPGA should not rely on a bootloader
to bring CPU1 out of reset. This aligns the SMP implementation that
is accepted by the Linux community for SOCFPGA.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
---
 arch/arm/mach-socfpga/core.h    | 3 +++
 arch/arm/mach-socfpga/platsmp.c | 8 ++------
 arch/arm/mach-socfpga/socfpga.c | 6 +-----
 3 files changed, 6 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-socfpga/core.h b/arch/arm/mach-socfpga/core.h
index 89f1a98..fedf1b4 100644
--- a/arch/arm/mach-socfpga/core.h
+++ b/arch/arm/mach-socfpga/core.h
@@ -42,4 +42,7 @@ extern unsigned long cpu1start_addr;
 
 #define SOCFPGA_SCU_VIRT_BASE   0xfffec000
 
+#define SOCFPGA_SCU_VIRT_BASE	0xfffec000
+#define SOCFPGA_SDMMC_BASE	0xff704000
+
 #endif /* __CORE_H */
diff --git a/arch/arm/mach-socfpga/platsmp.c b/arch/arm/mach-socfpga/platsmp.c
index 364951d..e4b567c 100644
--- a/arch/arm/mach-socfpga/platsmp.c
+++ b/arch/arm/mach-socfpga/platsmp.c
@@ -29,10 +29,6 @@
 
 #include "core.h"
 
-int pen_release = -1;
-
-static DEFINE_SPINLOCK(boot_lock);
-
 void __cpuinit platform_secondary_init(unsigned int cpu)
 {
 	/*
@@ -79,8 +75,8 @@ static void __init socfpga_smp_init_cpus(void)
 
 	/* sanity check */
 	if (ncores > num_possible_cpus()) {
-		pr_warn("socfpga: no. of cores (%d) greater than configured"
-			"maximum of %d - clipping\n", ncores, num_possible_cpus());
+		pr_warn("# of cores (%d) greater maximum of %d\n",
+			ncores, num_possible_cpus());
 		ncores = num_possible_cpus();
 	}
 
diff --git a/arch/arm/mach-socfpga/socfpga.c b/arch/arm/mach-socfpga/socfpga.c
index 1a97d83..46542f7 100644
--- a/arch/arm/mach-socfpga/socfpga.c
+++ b/arch/arm/mach-socfpga/socfpga.c
@@ -29,16 +29,12 @@
 
 #define SOCFPGA_NR_IRQS		512
 
+void __iomem *socfpga_scu_base_addr = ((void __iomem *)(SOCFPGA_SCU_VIRT_BASE));
 void __iomem *sys_manager_base_addr;
 void __iomem *rst_manager_base_addr;
 void __iomem *clk_mgr_base_addr;
 unsigned long cpu1start_addr;
 
-#define SOCFPGA_SCU_VIRT_BASE	0xfffec000
-#define SOCFPGA_SDMMC_BASE	0xff704000
-
-void __iomem *socfpga_scu_base_addr = ((void __iomem *)(SOCFPGA_SCU_VIRT_BASE));
-
 static const struct of_dev_auxdata socfpga_auxdata_lookup[] __initconst = {
 #ifdef CONFIG_MMC_DW
 	OF_DEV_AUXDATA("snps,dw-mmc", SOCFPGA_SDMMC_BASE, "snps,dw-mmc",
-- 
1.9.1

